fre/app/generate_time_averages/frepytoolsTimeAverager.py

```diff
@@ -14,8 +14,19 @@ class inheriting from abstract base class timeAverager
     '''
 
     def generate_timavg(self, infile=None, outfile=None):
-        ''' frepytools approach in a python-native manner.
-        deliberately avoids pre-packaged routines. '''
+
+        """
+        frepytools approach in a python-native manner.
+        deliberately avoids pre-packaged routines.
+
+        :param self: 
```

Comment:
same comment on the param self as the previous 2 files, noting that you shouldin this file refer to it as an instance of frepytoolsTimeAverager class